### PR TITLE
Telemetry on startup/shutdown

### LINKED ISSUE #342: Startup and shutdown telemetry
Use [vscode-commons](https://github.com/redhat-developer/vscode-commons) to collect opt-in telemetry for the extension.

Emit startup and shutdown telemetry events for the extension.

### PR DESCRIPTION
**Pull request by datho7561**
Use vscode-commons to collect telemetry data. Telemetry collection is opt-in. Please refer to `USAGE_DATA.md` in this PR for a comprehensive list of what data is collected.

Closes #342

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,6 +77,12 @@ The following settings are supported:
 Since 1.6.0:
 * `quarkus.tools.propertiesLanguageMismatch` : Action performed when detected Quarkus properties have an incorrect language. Default is `forceQuarkus`.
 
+## Telemetry
+
+vscode-quarkus extension collects anonymous [usage data](USAGE_DATA.md) and sends it to Red Hat servers to help improve our products and services.
```
Comment (rgrunber) [inline README.md]: I would add "With your approval, vscode-quarkus extension collects..."

**Review by rgrunber — CHANGES_REQUESTED** [review]
Change looks good overall though.

**Review by rgrunber — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -77,6 +77,12 @@ The following settings are supported:
 Since 1.6.0:
 * `quarkus.tools.propertiesLanguageMismatch` : Action performed when detected Quarkus properties have an incorrect language. Default is `forceQuarkus`.
 
+## Telemetry
+
+With your approval, vscode-quarkus extension collects anonymous [usage data](USAGE_DATA.md) and sends it to Red Hat servers to help improve our products and services.
+Read our [privacy statement](https://developers.redhat.com/article/tool-data-collection) to learn more.
+This extension respects the `redhat.elemetry.enabled` setting, which you can learn more about at https://github.com/redhat-developer/vscode-commons#how-to-disable-telemetry-reporting
```
Comment (rgrunber) [inline README.md:84]: @datho7561  Just noticed `elemetry` looking through the vscode-microprofile one. Should correct before releasing.

File: USAGE_DATA.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# USAGE DATA
+
+vscode-quarkus has opt-in telemetry collection, provided by [vscode-commons](https://github.com/redhat-developer/vscode-commons).
+
+## What's included in the vscode-xml telemetry data
```
Comment (rgrunber) [inline USAGE_DATA.md:5]: `vscode-xml` here as well.

File: README.md (hunk 2 above)
Comment (datho7561) [reply README.md:84]: #353

**Pull request merged**